receiver/apachesparkreceiver/scraper.go

```diff
@@ -0,0 +1,339 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+
+package apachesparkreceiver // import "github.com/open-telemetry/opentelemetry-collector-contrib/receiver/apachesparkreceiver"
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"time"
+
+	"go.opentelemetry.io/collector/component"
+	"go.opentelemetry.io/collector/pdata/pcommon"
+	"go.opentelemetry.io/collector/pdata/pmetric"
+	"go.opentelemetry.io/collector/receiver"
+	"go.opentelemetry.io/collector/receiver/scrapererror"
+	"go.uber.org/zap"
+	"golang.org/x/exp/slices"
+
+	"github.com/open-telemetry/opentelemetry-collector-contrib/receiver/apachesparkreceiver/internal/metadata"
+	"github.com/open-telemetry/opentelemetry-collector-contrib/receiver/apachesparkreceiver/internal/models"
+)
+
+var (
+	errClientNotInit             = errors.New("client not initialized")
+	errFailedAppIDCollection     = errors.New("failed to retrieve app ids")
+	errNoMatchingWhitelistedApps = errors.New("no apps matched whitelisted names")
+)
+
+type sparkScraper struct {
+	client   client
+	logger   *zap.Logger
+	config   *Config
+	settings component.TelemetrySettings
+	mb       *metadata.MetricsBuilder
+}
+
+func newSparkScraper(logger *zap.Logger, cfg *Config, settings receiver.CreateSettings) *sparkScraper {
+	return &sparkScraper{
+		logger:   logger,
+		config:   cfg,
+		settings: settings.TelemetrySettings,
+		mb:       metadata.NewMetricsBuilder(cfg.MetricsBuilderConfig, settings),
+	}
+}
+
+func (s *sparkScraper) start(_ context.Context, host component.Host) (err error) {
+	httpClient, err := newApacheSparkClient(s.config, host, s.settings)
+	if err != nil {
+		return fmt.Errorf("failed to start: %w", err)
+	}
+	s.client = httpClient
+	return nil
+}
+
+func (s *sparkScraper) scrape(_ context.Context) (pmetric.Metrics, error) {
+	now := pcommon.NewTimestampFromTime(time.Now())
+	var scrapeErrors scrapererror.ScrapeErrors
+
+	if s.client == nil {
+		return pmetric.NewMetrics(), errClientNotInit
+	}
+
+	// call applications endpoint to get ids and names for all apps in the cluster
```

Comment:
```suggestion
	// Call applications endpoint to get ids and names for all apps in the cluster
```
Golang comments should be capitalized